xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java

```diff
@@ -0,0 +1,228 @@
+      List<EquivalentAddressGroup> servers, Attributes attributes) {
+    Map<String, Object> newLbConfig = checkNotNull(
+        attributes.get(ATTR_LOAD_BALANCING_CONFIG), "ATTR_LOAD_BALANCING_CONFIG not available");
+    if (!newLbConfig.equals(lbConfig)) {
+      handleNewConfig(newLbConfig);
+    }
+    xdsLbState.handleResolvedAddressGroups(servers, attributes);
+  }
+
+  private void handleNewConfig(Map<String, Object> newLbConfig) {
+    String newBalancerName = ServiceConfigUtil.getBalancerNameFromXdsConfig(newLbConfig);
+    Map<String, Object> fallbackPolicy = selectFallbackPolicy(newLbConfig);
+    if (lbConfig == null) {
+      updateXdsLbState(
+          helper, newBalancerName, selectChildPolicy(newLbConfig), fallbackPolicy, null, null);
+    } else if (!newBalancerName.equals(
+        ServiceConfigUtil.getBalancerNameFromXdsConfig(lbConfig))) {
+      xdsLbState.shutdown();
+      xdsLbState.shutdownLbComm();
+      updateXdsLbState(
+          helper, newBalancerName, selectChildPolicy(newLbConfig), fallbackPolicy, null, null);
+    } else if (!Objects.equals(
+        ServiceConfigUtil.getChildPolicyFromXdsConfig(newLbConfig),
+        ServiceConfigUtil.getChildPolicyFromXdsConfig(lbConfig))) {
+      XdsLbState.Mode currentMode = xdsLbState.mode();
+      Map<String, Object> childPolicy = selectChildPolicy(newLbConfig);
+      Status cancel = Status.CANCELLED.withDescription("Changing loadbalancing mode");
+      switch (currentMode) {
+        case STANDARD:
+          if (childPolicy != null) {
+            // GOTO CUSTOM mode, close the stream but reuse the channel
+            xdsLbState.shutdown();
+            xdsLbState.shutdownLbRpc(cancel);
+            updateXdsLbState(
+                helper, newBalancerName, childPolicy, fallbackPolicy, xdsLbState.lbCommChannel(),
+                null);
+          } // else, still STANDARD mode
+          break;
+        case CUSTOM:
+          if (childPolicy != null) {
+            // GOTO a new CUSTOM mode, close the stream but reuse the channel
+            xdsLbState.shutdown();
+            xdsLbState.shutdownLbRpc(cancel);
+            updateXdsLbState(
+                helper, newBalancerName, childPolicy, fallbackPolicy, xdsLbState.lbCommChannel(),
+                null);
+          } else {
+            // GOTO STANDARD mode, reuse the stream
+            updateXdsLbState(
+                helper, newBalancerName, null, fallbackPolicy, xdsLbState.lbCommChannel(),
+                xdsLbState.adsStream());
+          }
+          break;
+        default:
+          throw new AssertionError("Unsupported xds plugin mode: " + currentMode);
+      }
+    }
+    lbConfig = newLbConfig;
+  }
+
+  private void updateXdsLbState(
+      Helper helper,
+      String balancerName,
+      @Nullable final Map<String, Object> childPolicy,
+      @Nullable Map<String, Object> fallbackPolicy,
+      @Nullable final ManagedChannel channel,
+      @Nullable final AdsStream adsStream) {
```

Comment:
Here you have one parameter per line which differs from the rest of the code. Wondering if there is a reason for this change of style

Reply:
Both styles are widely scattered in the code base. The other style is to put as many params as possible in one line, and it saves number of lines. Here, the params list may change frequently later on, so the one param per line style will be easier to modify the params list.

Reply:
I differ a bit and see 2 issues. The different style can become distracting while doing code reviews or even understanding code. Also things like this are typically captured and enforced in a project's styleguide - not sure if that's the case for the grpc-java styleguide.

Reply:
But I am okay with this and this is not a big issue for me